### ISSUE TITLE
`Either` deserializers `Option[T]` with value None as `null`

### ISSUE THREAD

**Issue by dope9967** [FasterXML/jackson-module-scala#472]
Jackson version 2.10.3.
Code to reproduce:
```
import com.fasterxml.jackson.module.scala.DefaultScalaModule

val right: Either[Option[String], Option[String]] = Right(None)
val left: Either[Option[String], Option[String]] = Left(None)

val mapper = new ObjectMapper()
mapper.registerModule(new DefaultScalaModule)

val rightString = mapper.writeValueAsString(right)
mapper.readValue(rightString, classOf[Either[Option[String], Option[String]]])

val leftString = mapper.writeValueAsString(left)
mapper.readValue(leftString, classOf[Either[Option[String], Option[String]]])
```

**Comment by dope9967** [comment]
Apparently this is covered in tests as valid behaviour - https://github.com/FasterXML/jackson-module-scala/blob/c29c39855439ed5822c568b1365f096cc3cd7e0f/src/test/scala/com/fasterxml/jackson/module/scala/deser/EitherDeserializerTest.scala#L29-L37. That basically makes `Either` containing `Option` not possible. Is this really correct?

**Comment by pjfanning** [comment]
@dope9967 those tests are not specifically about Either wrapping Options - this looks like a use case that was not considered and it should ideally be fixed

Would you be able to try producing a Pull Request to fix this? 2.10 will not have any more releases. 2.11 is probably not a good place for a big change. I would suggest targeting the 2.12 branch.

**Comment by dope9967** [comment]
@pjfanning sure, I can give it a try.

**Comment by pjfanning** [comment]
should be fixed in 2.12.0-SNAPSHOT now - I will backport to 2.11.3-SNAPSHOT in near future
